terraform/modules/deploy_hook/lambdas/hook.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+import os
+from datetime import datetime
+from http.client import HTTPResponse
+from urllib import request
+
+import boto3
+
+from common import get_auth_token, record_metric
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+gdx_url = os.environ["gdx_url"]
+events_url = gdx_url + "/events"
+
+auth_url = os.environ["auth_url"]
+
+client_id = os.environ["client_id"]
+client_secret = os.environ["client_secret"]
+
+codedeploy = boto3.client('codedeploy')
+
+
+def lambda_handler(event, _context):
+    logger.info(f"## EVENT: {event}")
+    logger.info(f"## TIME: {datetime.now()}")
+
+    put_lifecycle_event_hook(event, 'InProgress')
+
+    auth_token = get_auth_token(auth_url, client_id, client_secret)
+
+    post_event(auth_token)
+    events = get_events(auth_token)
+    get_event(auth_token, events[0]["id"])
+
+    put_lifecycle_event_hook(event, 'Succeeded')
```

Comment:
Do we need to do error handling and explicitly put a failed event?

Reply:
Yes I should do that